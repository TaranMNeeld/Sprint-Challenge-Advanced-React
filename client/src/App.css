.App {
  text-align: center;
}

.toggle-div {
  width: 80px;
  height: 40px;
}

.toggle-btn {
  width: 70px;
  height: 35px;
}

.inverted-color {
  background: black;
  color: white;
}