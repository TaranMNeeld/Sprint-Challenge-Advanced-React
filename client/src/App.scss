.App {
  text-align: center;
}

.inverted-color {
  background: black;
  color     : white;

  * .player-card {
    border    : 1px solid white;
  }
}