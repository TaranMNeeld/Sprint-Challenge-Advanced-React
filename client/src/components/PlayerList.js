import React from 'react';
import axios from 'axios';
import PlayerCard from './PlayerCard';
import '../styles/PlayerList.css'

class PlayerList extends React.Component {
  constructor() {
    super();
    this.state = {
      players: []
    };
  };

  componentDidMount() {
    axios.get('http://localhost:5000/api/players')
      .then(res => this.setState({
          players: res.data
      }))
      .catch(err => console.log(err));
  }

  render() {
    return (
      <div className='player-list'>
          {this.state.players.map(player => {
              return <PlayerCard key={player.id} name={player.name} country={player.country} />
          })}
      </div>
    );
  }
}

export default PlayerList;