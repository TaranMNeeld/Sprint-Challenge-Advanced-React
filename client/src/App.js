import React from 'react';
import './App.scss';
import PlayerList from './components/PlayerList';
import {useInvertedColors} from './hooks/useInvertColors';

function App() {

  const [invert, setInvert] = useInvertedColors(false);

    const toggle = e => {
        e.preventDefault();
        setInvert(!invert);
    }

  return (
    <div className='App'>
      <button className='toggle-btn' onClick={toggle}>Invert Colors</button>
      <PlayerList/>
    </div>
  );
}

export default App;
