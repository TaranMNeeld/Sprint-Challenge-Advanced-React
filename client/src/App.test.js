import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import PlayerList, {apiData} from './components/PlayerList';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('gets an array from api', () => {
  expect(typeof apiData).toBe('Array');
});