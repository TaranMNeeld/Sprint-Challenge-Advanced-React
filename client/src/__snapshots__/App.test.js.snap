// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlayerList component 1`] = `undefined`;
